src/librustc_typeck/check/cast.rs

```diff
@@ -365,28 +365,34 @@ impl<'a, 'gcx, 'tcx> CastCheck<'tcx> {
     fn trivial_cast_lint(&self, fcx: &FnCtxt<'a, 'gcx, 'tcx>) {
         let t_cast = self.cast_ty;
         let t_expr = self.expr_ty;
+        let type_asc_or = if fcx.tcx.features().type_ascription {
+            "type ascription or "
+        } else {
+            ""
+        };
         if t_cast.is_numeric() && t_expr.is_numeric() {
```

Comment:
The only difference between the two branches is the lint ID and the word `numeric`, can you pull those out into variables and only make a single error emitting call?

Reply:
done